Documentation/Communicational Patterns/2. Tooling/2.4. ObservedObject.md

```diff
@@ -0,0 +1,46 @@
+![document type: vision](https://themomax.github.io/resources/markdown-labels/document_type_vision.svg)
+
+# ObservedObject
+
+The `@ObservedObject` wraps an `ObservableObject`. An `ObservableObject` may have multiple `@Published` properties. Each `@Published` property is projected to a `Publisher`. A `Handler` that contains an `@ObservedObject` subscribes to these `Publisher`s and evaluates the `Handler` on each update.
+
+## Lifetime
+
+### Influence
+
+The presence of `@ObservedObject` properties on a `Handler` signalizes exporting the endpoint as a Server-Side Stream makes sense. If the exporter decides to do so the lifetime of the `Handler` has to be extended accordingly. The `Handler` should stay alive until all `Publisher`s projected from observed properties have finished.
```

Comment:
I think that the idea of having to wait for all `@Published` properties `Publisher`s are finished is a approach that sounds a bit complicated to me. When we take a look at SwiftUI we don't have the notion of closing an `ObservedObject`'s publisher or the publishers of the properties marked with `@Published`.
I would suggest that the decision of when a `Handler` should close the service side or bidirectional stream should be decided within the handler and within the `handle` method. I would propose to add an additional `.end(Reponse)` `Action` that behaves the same as `.send(Reponse)` for all. communication patters except for service-side and bidirectional streams.

Reply:
duplicate